src/lib/OccAnalysis.hs

```diff
@@ -300,11 +289,23 @@ occNest a decls ans = case decls of
         let binding'' = DeclBinding (OccInfo usage) ty expr
         return $ Abs (Nest (Let b' binding'') ds'') ans''
 
+checkAllFreeVariablesMentioned :: HoistableE e => e n -> OCCM n ()
+checkAllFreeVariablesMentioned e = do
+#ifdef DEX_DEBUG
+  FV fvs <- get
+  forM_ (nameSetToList (freeVarsE e)) \name ->
+    case lookupNameMapE name fvs of
+      Just _ -> return ()
+      Nothing -> error $ "Free variable map missing free variable " ++ show name
+#else
+  void $ return e  -- Refer to `e` in this branch to avoid a GHC warning
+  return ()
```

Comment:
Add `INLINE` in the `else` branch to make sure this gets fully eliminated

Reply:
Done.